backend/main.cpp

```diff
@@ -0,0 +1,199 @@
+}
+
+void handle_request(const std::string& request, ip::tcp::socket& socket) {
+    try {
+
+        std::cout << "Request received: " << request << std::endl;
+        // Handle OPTIONS request
+        if (request.substr(0, 7) == "OPTIONS") {
+            handle_options_request(socket);
+            return;
+        }
+
+        else if (request.substr(0, 4) != "POST") {
+            // Send a 405 Method Not Allowed response
+            std::string response = "HTTP/1.1 405 Method Not Allowed\r\n";
+            response += "Content-Type: text/plain\r\n";
+            response += "Access-Control-Allow-Origin: *\r\n"; // Enable CORS for all origins
+            response += "Content-Length: 0\r\n\r\n";
+            boost::asio::write(socket, boost::asio::buffer(response));
+            return;
+        }
+
+        if (request.empty()) {
+            // std::cout << "Empty request received." << std::endl;
+            throw std::invalid_argument("Empty request received.");
+            std::cout << "Empty request received." << std::endl;
+            return;
+        }
+
+        // Find the position of the first '{' character after the headers
+        size_t json_start_pos = request.find("{");
+        if (json_start_pos == std::string::npos) {
+            throw std::invalid_argument("JSON data not found in request.");
+        }
+
+        // Extract the JSON payload from the request
+        std::string json_payload = request.substr(json_start_pos);
+
+        // Parse the JSON payload into the Matrices struct
+        auto matrices_payload = nlohmann::json::parse(json_payload)["matrices"];
+
+        std::cout << "Matrices received successfully!" << std::endl;
+
+        Matrices matrices(matrices_payload[0], matrices_payload[1]);
+
+        // Print input matrices as sanity check
+        std::cout << "Input Matrix 1:" << std::endl;
+        for (const auto& row : matrices.input_matrix_1) {
+            for (const auto& cell : row) {
+                std::cout << cell << " ";
+            }
+            std::cout << std::endl;
+        }
+        
+        std::cout << "Input Matrix 2:" << std::endl;
+        for (const auto& row : matrices.input_matrix_2) {
+            for (const auto& cell : row) {
+                std::cout << cell << " ";
+            }
+            std::cout << std::endl;
+        }
+
+        // CONVERT FLOATS TO FPGA/HARDWARE FRIENDLY FORMAT, PROBABLY INTS??
+        // TALK WITH FPGA AND GET RESULTS
+        // SCALE RESULTS "12-bit dac with 4.096 reference voltage -> analog switch with 15ohms of resistance 
+        // -> analog multipler that divides the result by a factor of 10 -> 16-bit adc with 4.096 reference voltage"
+
+        // UPDATE RESULT MATRIX
+        // TODO: REMOVE THIS DUMMY OPERATION
+        // Multiply matrices 1 and 2 and store the result in the result matrix
+        // This is a leetcode question I think LOL
+        for (int i = 0; i < matrices.input_matrix_1.get_num_cols(); i++) {
```

Comment:
Note for me to implement later: Add compiler directives eventually so we can leave these things in as debug operations